server/src/test/java/org/elasticsearch/index/mapper/LongSyntheticSourceNativeArrayIntegrationTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper;
+
+import com.carrotsearch.randomizedtesting.generators.RandomStrings;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LongSyntheticSourceNativeArrayIntegrationTests extends NativeArrayIntegrationTestCase {
+
+    @Override
+    protected String getFieldTypeName() {
+        return "long";
```

Comment:
Maybe create subclasses here for the other number field types?

Reply:
I think subclasses for other number types are still missing?

Reply:
Sorry, I'm still working on getting those in (I had a bit of trouble with types when checking the expected vs. output results). I'll be sure to get those in before merging